ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx

```diff
@@ -289,7 +308,23 @@ function WorkloadCvesOverviewPage() {
         return apolloClient.refetchQueries({ include: [imageListQuery] });
     }
 
-    const filterToolbar = (
+    const filterToolbar = isAdvancedFiltersEnabled ? (
+        <AdvancedFiltersToolbar
+            className="pf-v5-u-py-md"
+            searchFilterConfig={searchFilterConfig}
+            searchFilter={searchFilter}
+            defaultFilters={localStorageValue.preferences.defaultFilters}
+            onFilterChange={(newFilter, { action }) => {
+                setSearchFilter(newFilter);
+                pagination.setPage(1, 'replace');
+
+                if (action === 'ADD') {
+                    // TODO - Add analytics tracking ROX-24532
+                }
```

Comment:
I wonder what are the pros/cons of specifying analytics at this level vs. within `CompoundSearchFilter`. I can see how this would be preferable If we want to opt out of analytics for advanced filters in certain contexts. I guess keeping it out of the lower-level component would also make it more apparent that it's integrated while putting it in the lower-level component might go unnoticed. The downside is that we would need to specify we want to integrate analytics each time we use Advanced Filters. I can see how that might become tedious.

Reply:
Yeah this is a good point, from what I can tell the main pros/cons of doing it this way are:

Pros:
- Flexibility to track, or how we want to track, values can be easily customizable at the Container level. e.g. In VM at least, we are tracking the **value** of some filters but for the most part are only tracking the category.
- We can more easily customize the context of where and why a filter was changed, like the page or subsection a filter was applied to.
- No need to have tracking specific props in the search component, we can have the two concerns (analytics and filtering) kept separate

Cons:
- We need to specify a tracking function at every call site, which could be tedious or easy to forget
- Potential for duplicate code

Could be worth discussing more at a team sync though.